src/CloudSecrets.UnitTests/ConnectedModeSecretsTests.cs

```diff
@@ -0,0 +1,118 @@
+        {
+            var sonarQubeService = new Mock<ISonarQubeService>();
+            sonarQubeService.Setup(x => x.IsConnected).Returns(false);
+
+            var testSubject = new ConnectedModeSecrets(sonarQubeService.Object);
+
+            var result = testSubject.AreSecretsAvailable();
+
+            sonarQubeService.Verify(x => x.IsConnected, Times.Once);
+            sonarQubeService.Verify(x => x.GetServerInfo(), Times.Never);
+            result.Should().BeFalse();
+        }
+
+        [TestMethod]
+        public void AreSecretsAvailable_ProjectIsConnectedToSonarCloud_Available()
+        {
+            var sonarQubeService = new Mock<ISonarQubeService>();
+            sonarQubeService.Setup(x => x.IsConnected).Returns(true);
+
+            var serverInfo = CreateServerInfo(ServerType.SonarCloud);
+            sonarQubeService.Setup(x => x.GetServerInfo()).Returns(serverInfo);
+
+            var testSubject = new ConnectedModeSecrets(sonarQubeService.Object);
+
+            var result = testSubject.AreSecretsAvailable();
+
+            sonarQubeService.Verify(x => x.IsConnected, Times.Once);
+            sonarQubeService.Verify(x => x.GetServerInfo(), Times.Once);
+            result.Should().BeTrue();
+        }
+
+        [TestMethod]
+        public void AreSecretsAvailable_ProjectIsConnectedToSonarQube_BelowRequiredVersion_NotAvailable()
+        {
+            var sonarQubeService = new Mock<ISonarQubeService>();
+            sonarQubeService.Setup(x => x.IsConnected).Returns(true);
+
+            var serverInfo = CreateServerInfo(ServerType.SonarQube, new Version(1, 5));
+            sonarQubeService.Setup(x => x.GetServerInfo()).Returns(serverInfo);
+
+            var testSubject = new ConnectedModeSecrets(sonarQubeService.Object);
+
+            var result = testSubject.AreSecretsAvailable();
+
+            sonarQubeService.Verify(x => x.IsConnected, Times.Once);
+            sonarQubeService.Verify(x => x.GetServerInfo(), Times.Once);
+            result.Should().BeFalse();
+        }
+
+        [TestMethod]
+        public void AreSecretsAvailable_ProjectIsConnectedToSonarQube_AboveRequiredVersion_Available()
+        {
+            var sonarQubeService = new Mock<ISonarQubeService>();
+            sonarQubeService.Setup(x => x.IsConnected).Returns(true);
+
+            var serverInfo = CreateServerInfo(ServerType.SonarQube, new Version(9, 9));
+            sonarQubeService.Setup(x => x.GetServerInfo()).Returns(serverInfo);
+
+            var testSubject = new ConnectedModeSecrets(sonarQubeService.Object);
+
+            var result = testSubject.AreSecretsAvailable();
+
+            sonarQubeService.Verify(x => x.IsConnected, Times.Once);
+            sonarQubeService.Verify(x => x.GetServerInfo(), Times.Once);
+            result.Should().BeTrue();
+        }
+
+        private ServerInfo CreateServerInfo(ServerType serverType, Version version = null)
```

Comment:
You could expand this method (or create another one that wraps it) that creates and returns the configure `SonarQubeService`. That would simplify the setup of each test quite a bit.